### DISCUSSION TITLE
请问有没有支持离线model的ASR、TTS的Python代码呢？

### DISCUSSION THREAD

**Discussion by qibinran** [PaddlePaddle/PaddleSpeech#1423] [Q&A]
有好几个讨论里面提到识别和合成速度慢的问题（#1256； #1280），问题主要是现有的CL每次都重新下载模型；已有的Python代码“ASRExecutor”（https://github.com/PaddlePaddle/PaddleSpeech/tree/develop/demos/speech_recognition）、“TTSExecutor”（https://github.com/PaddlePaddle/PaddleSpeech/tree/develop/demos/text_to_speech）也会每次重新下载模型。如果将模型下载下来，请问是否有从本机调用模型、且只调用一次的ASR、TTS的Python代码呢？

**Comment by yt605155624** [comment]
模型只会在第一次调用的时候下载😊

**Comment by qibinran** [comment]
谢谢回复！前面写得不准确，是第一次会下载模型，每次都会加载模型。按#1177 ，Wenet只调用一次模型，paddlespeech慢4倍。请问有没有从本机调用模型、且只调用一次的ASR、TTS的Python代码呢？

**Comment by yt605155624** [comment]
类似的功能在开发中了

**Reply by qibinran** [reply]
谢谢！
